emgapianns/management/commands/clean_study.py

```diff
@@ -18,105 +18,151 @@
-                '3.0',
-                '4.0',
-                '4.1',
-                '5.0'],
-            default='5.0'
+            "--pipeline",
+            help="Pipeline version",
+            action="store",
+            dest="pipeline",
+            choices=["1.0", "2.0", "3.0", "4.0", "4.1", "5.0"],
+            default="5.0",
         )
 
     def handle(self, *args, **options):
-
+        """Execute the command"""
         pipeline_version = options.get("pipeline")
         study_id = int(options.get("accession").replace("MGYS", ""))
-        study = Study.objects.get(pk=study_id)
 
-        download_group = options.get("download_group")
-        selected_groups = self.groups[download_group]
-        groups = DownloadGroupType.objects.filter(group_type__in=selected_groups)
+        try:
+            study = Study.objects.get(pk=study_id)
+        except Study.DoesNotExist:
+            logger.error("Study " + str(study_id) + " doesn't exist")
+            return
+
+        download_groups = options.get("group")
+
+        selected_groups = []
+        if "all" in download_groups:
+            groups = list(DownloadGroupType.objects.all())
+        else:
+            groups = list(
+                g
+                for g in DownloadGroupType.objects.all()
+                if _slugify(g.group_type) in download_groups
+            )
 
         # Get all the Analysis
-        jobs = AnalysisJob.objects.filter(study=study, pipeline__release_version=pipeline_version)
+        jobs = AnalysisJob.objects.filter(
+            study=study, pipeline__release_version=pipeline_version
+        )
+
+        logger.info("Analyses jobs to remove: " + str(jobs.count()))
 
         for job in jobs:
-            logger.info("Cleaning job {}".format(job))
+            logger.info("Cleaning analysis job {}".format(job))
+
             downloads = AnalysisJobDownload.objects.filter(
-                job=job, group_type__in=groups,
-                pipeline__release_version=pipeline_version)
+                job=job,
+                group_type__in=groups,
+                pipeline__release_version=pipeline_version,
+            )
 
             if downloads:
-                logger.info("Found {} downloads for job: {}".format(len(downloads), job))
+                logger.info(
+                    "Found {} downloads for analysis job: {}".format(
+                        len(downloads), job
+                    )
+                )
                 logger.info("Removing the following files:")
                 [logger.info(d.realname) for d in downloads]
-                [d.delete() for d in downloads]
+                downloads.delete()
                 logger.info("Removed.")
             else:
                 logger.info("No files to remove.")
 
-            taxonomy_annotations = AnalysisJobTaxonomy.objects.\
-                filter(job_id=job.job_id, pipeline_version=pipeline_version)
+            if not options.get("mongo_tax"):
```

Comment:
Does the next section need an else for mongo_tax=True?

Reply:
No, it's an optional flag. The user has to specify the `-m` flag otherwise the value `mongo_tax` will be None.